### PR TITLE
mgr/dashboard: fix issues related with PyJWT versions >=2.0.0

### PR DESCRIPTION
**Pull request by alfonsomthd**
Fixes: https://tracker.ceph.com/issues/49574
Signed-off-by: Alfonso Martínez <almartin@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by alfonsomthd** [comment]
jenkins test make check

**Comment by alfonsomthd** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/controllers/auth.py [hunk 1]
```diff
@@ -52,7 +52,8 @@ def create(self, username, password):
                 mgr.ACCESS_CTRL_DB.reset_attempt(username)
                 mgr.ACCESS_CTRL_DB.save()
                 token = JwtManager.gen_token(username)
-                token = token.decode('utf-8')
+                # For backward-compatibility with PyJWT versions < 2.0.0
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/controllers/auth.py:55]: Please specify which version is returning bytes and which str.

File: src/pybind/mgr/dashboard/constraints.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 CherryPy==13.1.0
 more-itertools==4.1.0
-PyJWT==1.6.4
+PyJWT==2.0.1
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/constraints.txt:3]: Just curious: did this catch the issue via pylint/unit tests? If Ubuntu Bionic is still using jwt 1.x, we can have a separate 'contraints-ubuntu-bionic.txt' and run pylint/unit-tests in a matrix style.

**Review by epuertat — APPROVED** [review]
Thanks @alfonsomthd, looks good to me!

Just a small suggestion on the comments over there, and an open reflection on how we can try to avoid this situation in the future:
* Manually updating `constraints-{distro}-{release}.txt` for each of the distros-releases supported and [using generative tox lists](https://tox.readthedocs.io/en/latest/config.html#generative-envlist) (testing matrix). However that requires manually updating the dependencies and wouldn't have helped in this case.
* Generating those `constraints.txt` files on the fly by fetching the package versions for every distro (via apt/yum/etc).
* Using docker images for every distro-release to run tox inside from site packages (instead of pip-installed ones).

File: src/pybind/mgr/dashboard/constraints.txt (hunk 2 above)
Comment (alfonsomthd) [reply src/pybind/mgr/dashboard/constraints.txt:3]: No, this was preventing Ceph API tests job from failing:
https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/run-backend-api-tests.sh#L49

File: src/pybind/mgr/dashboard/controllers/auth.py (hunk 1 above)
Comment (alfonsomthd) [reply src/pybind/mgr/dashboard/controllers/auth.py]: Done.

File: src/pybind/mgr/dashboard/constraints.txt (hunk 2 above)
Comment (alfonsomthd) [reply src/pybind/mgr/dashboard/constraints.txt:3]: I provided the python unit test that would have caught this issue.

**Comment by epuertat** [comment]
jenkins test dashboard

**Comment by liewegas** [comment]
https://pulpito.ceph.com/sage-2021-03-03_15:57:18-rados:dashboard-wip-sage-testing-2021-03-03-0847-distro-basic-gibba/ :100:

**Review by avanthakkar — APPROVED** [review]

**Pull request merged**